### PR TITLE
[rllib] Fixed num_atoms>1 in pytorch

### PR DESCRIPTION
**Pull request by holli**
## Why are these changes needed?

There is a bug when handling num_atoms>1 with dqn with pytorch.

Running e.g. https://github.com/ray-project/ray/blob/master/rllib/tuned_examples/dqn/atari-dist-dqn.yaml , with torch and SpaceInvadersNoFrameskip-v4 env resulted minimal reward after around 30 iterations.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by holli** [comment]
A sanity check and a example for reviewer:

```
support_logits_per_action = torch.zeros(1, 3, 5).uniform_()
nn.functional.softmax(support_logits_per_action)
nn.functional.softmax(support_logits_per_action, dim=0)
nn.functional.softmax(support_logits_per_action, dim=1)
nn.functional.softmax(support_logits_per_action, dim=2)
nn.functional.softmax(support_logits_per_action, dim=-1)

== Results to ==>

tensor([[[1., 1., 1., 1., 1.],
         [1., 1., 1., 1., 1.],
         [1., 1., 1., 1., 1.]]])
tensor([[[1., 1., 1., 1., 1.],
         [1., 1., 1., 1., 1.],
         [1., 1., 1., 1., 1.]]])
tensor([[[0.4154, 0.3106, 0.3641, 0.3092, 0.3015],
         [0.2569, 0.3682, 0.3329, 0.2231, 0.4931],
         [0.3277, 0.3212, 0.3030, 0.4677, 0.2054]]])
tensor([[[0.2841, 0.2074, 0.1729, 0.1770, 0.1586],
         [0.1818, 0.2544, 0.1635, 0.1321, 0.2683],
         [0.2339, 0.2239, 0.1501, 0.2794, 0.1128]]])
tensor([[[0.2841, 0.2074, 0.1729, 0.1770, 0.1586],
         [0.1818, 0.2544, 0.1635, 0.1321, 0.2683],
         [0.2339, 0.2239, 0.1501, 0.2794, 0.1128]]])

```

**Comment by holli** [comment]
Not really relevant to this pull request but I was wondering are those examples ran with pytorch in some tests or somewhere? E.g. in https://docs.ray.io/en/latest/rllib-algorithms.html#dqn would be nice to have pytorch and tf versions side by side to make sure that both work as intented and that there are no performance (bug) differences.

**Comment by ericl** [comment]
Thanks for looking at this @holli . We run a limited set of large scale regression tests here: https://github.com/ray-project/ray/blob/master/rllib/tuned_examples/compact-regression-test.yaml

Unfortunately, the coverage of torch/advanced options like distributional dqn isn't quite up to par yet, though we are working to improve the testing coverage and run them more often instead of once per release.

**Review by ericl — APPROVED** [review]

**Comment by holli** [comment]
@ericl ok cool there those tests were. Those kind of test would have captured this bug. From a user's point of view a wider set once per release is enough :) From a developers point of view sure more often would be nicer.

If I look at that file and the results. There is only one `framework: torch` there. The last results are from 0.8.6 (https://github.com/ray-project/ray/blob/master/doc/dev/release_logs/0.8.6/rllib_regression.txt#L23)... There are 4 PPO_BreakoutNoFrameskip-v4_926d2_00004 for tensorflow achieving reward≈30, and then 4 for pytorch way too slow achieving only reward≈3.  Probably hard to notice because the trial name doesn't show the test specifics. I would guess that's some gpu usage issue and PPO is working nowadays. 

In any case that kind of comparing tf -> pytorch I was looking for, just to make sure that newer pytorch versions of the algorithms are working.

**Comment by holli** [comment]
Ach the 0.8.7 release was found under the tag name and not in the master branch: https://github.com/ray-project/ray/blob/ray-0.8.7/doc/dev/release_logs/0.8.7/rllib_regression.txt#L23 . Pytorch PPO speed about half of the TF version. 

So yeah, few more tests and torch versions from all applicable would be nice. And are the trials easy to name so that they would reflect the intended test a bit more. Maybe in some another issue.
